### DISCUSSION TITLE
Is it possible to eliminate my duplicated code and make it work on multiple model types?

### DISCUSSION THREAD

**Discussion by narsimhamchelluri-ds** [aarondl/sqlboiler#1251] [Q&A] [answered]
Suppose I have two models `Cat` and `Dog` and both of them have a `Name string` field.

I am wondering if there is a way I can write a function in my own code that allows me to pass in either a `Cat` or a `Dog` instance (or a pointer to such an instance) and access the `Name` field within whatever was passed in.

I thought that it might be possible to do this if there was a way that the generated model code provided accessor methods for the fields like

```golang
func (c Cat) GetName() string {
	return c.Name
}
```

or maybe

```golang
func (c *Cat) GetName() string {
        if c == nil {
          return ""
        }
	return c.Name
}
```

If similar methods were generated for the `Dog` type, then I could create an interface like

```golang
type HasName interface {
    GetName() string
}
```

that was implemented by both of my model types, and then I could make a function like

```golang
func doStuff(n HasName) {
    fmt.Printf("Hello I have name %s!\n", n.GetName())
}
```

I am not sure that this approach is correct or that it is possible using the current model generation but I don't see it as being available. Instead I have to write two pretty-nearly identical funcs, one that operates on each type:

```golang
func doCatStuff(c models.Cat) {
    fmt.Printf("Hello I have name %s!\n", c.Name)
}

func doDogStuff(d models.Dog) {
    fmt.Printf("Hello I have name %s!\n", d.Name)
}
```

Is there a way around this code duplication?

Thank you for your help. I am still getting my feet wet with SQLBoiler and it is quite possible that I am missing something.

**Comment by stephenafamo** [comment, marked as answer]
You could pass custom templates to SQLBoiler to generate these, but there is little documentation on how to write SQLBoiler templates.

**Reply by narsimhamchelluri-ds** [reply]
I see, thank you. Perhaps if I have time later I will make an effort to do that. If I am able to make headway, perhaps I will make some notes, and those could be the seeds for some docs on templates.

**Reply by stephenafamo** [reply]
Thank you.
